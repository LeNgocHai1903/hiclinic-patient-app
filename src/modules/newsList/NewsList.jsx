import { useEffect, useState } from 'react';
import './NewsList.scss';
import { useTranslation } from 'react-i18next';

import MainSearchBar from '../../components/searchBar/mainSearchBar/MainSearchBar';
import NewsItem from '../../modules/newsList/newsItem/NewsItem';
import LoadingSpinner from '../../components/loadingSpinner/LoadingSpinner';
import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
import apiWrapper from '../../api/apiWrapper';
import * as sortType from '../../constant/sorting/sorting';
import * as variableType from '../../constant/variable/variableNumber';

const NewsList = (props) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);
  const [changePage, setChangePage] = useState(0);
  const [numberOfPages, setNumberOfPages] = useState(null);
  const { t } = useTranslation();

  useEffect(() => {
    apiWrapper({
      url: `${process.env.REACT_APP_PATIENT_NEWS_SERVER}/?size=${variableType.NUMBER_OF_PAGE_NEWS_LIST}&page=${changePage}`,
      method: 'GET',
    }).then((res) => {
      setData(res);
      setNumberOfPages(res.numberOfPage);
      setLoading(false);
    });
  }, [changePage]);

  const changeSortHandler = (e) => {
    switch (e.target.value) {
      case sortType.SORT_BY_ALPHABET_INC:
        setData([...data].sort((a, b) => a.title.localeCompare(b.title)));
        break;
      case sortType.SORT_BY_ALPHABET_DECS:
        setData([...data].sort((a, b) => a.title.localeCompare(b.title)).reverse());
        break;
      default:
        return setData([...data]);
    }
  };

  const displayPagination = [];

  const changePageHandler = (index) => {
    setChangePage(index);
  };

  if (numberOfPages) {
    for (let i = 0; i < numberOfPages; i++) {
      console.log(i);
      displayPagination.push(
        <PaginationItem active={i === changePage}>
          <PaginationLink onClick={() => changePageHandler(i)}>{i + 1}</PaginationLink>
        </PaginationItem>,
      );
      console.log(i);
    }
  }

  return (
    <div>
      <>
        {loading ? (
          <div className="loading">
            <LoadingSpinner />
          </div>
        ) : (
          <div>
            <MainSearchBar />
            <h1>{t('newsList')}</h1>
            <div className="news-sort">
              <select id="sort-list" onChange={changeSortHandler} className="sort-option">
                <optgroup label="Alphabet">
                  <option value="alphabet inc"> {t('a_z')} </option>
                  <option value="alphabet desc"> {t('z_a')} </option>
                </optgroup>
              </select>
            </div>
            {/* <Pagination data={data} type={'news'} itemPerPage={3}>
              <NewsItemGrid />
            </Pagination> */}
            {console.log('data: ', data)}
            {data.listNews.map((item) => {
              return <NewsItem data={item} />;
            })}
            <Pagination className="list-pagination">
              <PaginationItem disabled={changePage === 0}>
                <PaginationLink previous onClick={() => changePageHandler(changePage - 1)} />
              </PaginationItem>
              {displayPagination}
              <PaginationItem disabled={changePage === numberOfPages - 1}>
                <PaginationLink next onClick={() => changePageHandler(changePage + 1)} />
              </PaginationItem>
            </Pagination>
          </div>
        )}
      </>
    </div>
  );
};

export default NewsList;
