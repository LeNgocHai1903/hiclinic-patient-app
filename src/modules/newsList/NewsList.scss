.loading {
  display: flex;
  justify-content: center;
}

h1 {
  display: flex;
  justify-content: center;
  padding: 10px;
}

.news-sort {
  display: flex;
  justify-content: flex-end;
  padding: 10px;
  margin-bottom: 20px;
  width: 90%;
}

.sort-option {
  height: 1.8rem;
  width: 4rem;
  background-color: white;
  border-radius: 0.5rem;
  outline: none;
}

.list-pagination {
  margin-top: 20px;
  display: flex;
  justify-content: center;
}
