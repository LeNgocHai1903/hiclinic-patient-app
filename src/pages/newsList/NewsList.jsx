import { useEffect, useState } from "react";
import "./NewsList.scss";

import axios from "axios";

import Footer from "../../components/footer/Footer";
import Header from "../../components/header/Header";
import NewsItem from "../../modules/news/NewsItem";
import LoaddingSpinner from "../../components/loadingSpinner/LoadingSpinner";
import Pagination from "../../components/pagination/Pagination";

const NewsList = (props) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    const fetchData = async () => {
      const res = await axios(
        "https://hiclinic-patient-portal-server.herokuapp.com/api/news"
      );
      setData(res.data);
      setLoading(false);
    };
    fetchData();
  }, []);

  const changeSortHandler = (e) => {
    if (e.target.value === "alphabet inc") {
      setData([...data].sort((a, b) => a.title.localeCompare(b.name)));
    } else {
      setData(
        [...data].sort((a, b) => a.title.localeCompare(b.name)).reverse()
      );
    }
  };

  return (
    <div>
      <>
        <Header />
        {loading ? (
          <div className="loading">
            <LoaddingSpinner />
          </div>
        ) : (
          <div>
            <h1>News List</h1>
            <div className="news-sort">
              <select
                id="sort-list"
                onChange={changeSortHandler}
                className="sort-option"
              >
                <optgroup label="Alphabet">
                  <option value="alphabet inc"> A - Z </option>
                  <option value="alphabet desc"> Z - A </option>
                </optgroup>
              </select>
            </div>
            <Pagination data={data} type={"news"} itemPerPage={6}>
              <NewsItem />
            </Pagination>
          </div>
        )}
        <Footer />
      </>
    </div>
  );
};

export default NewsList;
