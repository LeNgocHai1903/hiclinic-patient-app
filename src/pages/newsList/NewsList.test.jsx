import React from 'react'
import renderer from 'react-test-renderer'
import { act } from 'react-dom/test-utils'
import { render, unmountComponentAtNode } from 'react-dom';
import NewsList from './NewsList'
import { MemoryRouter } from 'react-router-dom';

describe('News List Component', () => {
    it('should render component correctly', () => {
        const component = renderer.create(<NewsList />).toJSON();
        expect(component).toMatchSnapshot();
    })

    it('should render component correctly', () => {
        const component = renderer.create(
            <MemoryRouter>
                <NewsList />
            </MemoryRouter>
        ).toJSON();
        expect(component).toMatchSnapshot();
    })
})