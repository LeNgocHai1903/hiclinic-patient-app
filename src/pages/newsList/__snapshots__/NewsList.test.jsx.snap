// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News List Component should render component correctly 1`] = `
<div>
  <div
    className="loading"
  >
    <div
      className="undefined"
    >
      <div
        className="lds-dual-ring"
      />
    </div>
  </div>
</div>
`;

exports[`News List Component should render component correctly 2`] = `
<div>
  <div
    className="loading"
  >
    <div
      className="undefined"
    >
      <div
        className="lds-dual-ring"
      />
    </div>
  </div>
</div>
`;
